### PR TITLE
Deposit natural denom

### PR DESCRIPTION
**Pull request by mhagel**
## Link to Issue
Closes: #4922

## Description of Changes
- Adds util functions to convert minimalDenom amount to/from naturalDenom amount
  - Ex: 8000000uosmo -> 8 OSMO
  - 7 EVMOS -> 7000000000000000000aevmos
- Adds a ETHERMINT_CHAINS constant to set those to 18 decimals (was at 6 before)
- Displays deposit input in _DENOM_
- converts user input to _udenom_ for transaction
- Updates ethermint devnet bootstrap.sh to use a more sane minDeposit of 20 EVMOS

## Test Plan
- Added unit tests for conversion functions
- CA (click around) tested on local and frack:
  - Using shared account, join /csdk-beta community
  - Click "+" to make new onchain proposal, Choose "Community Spend"
    - Confirm Deposit field defaults to 2 STAKE
    - Confirm Amount field is denominated in STAKE
  - Create a proposal with `cosmos18q3tlnx8vguv2fadqslm7x59ejauvsmnlycckg` as recipient, and **0.000005** STAKE as the "Amount." Submit with keplr.
    - When the TX goes through, confirm that the processed spend amount is correct by expanding the data object in the "Community Spend Proposal" section of the proposal page
      - Confirm the amount is 5 ustake:
      - Screenshot 2023-09-07 at 4 27 28 PM

  - check `/evmos/new/proposal` (3500 EVMOS) or `/evmos-dev/new/proposal` (20 EVMOS) to confirm correct denoms

## Deployment Plan

1. 

## Other Considerations

- Most Cosmos chains use 6 decimal places for their minimal denom. Ethermint chains use 18.

### CODE REVIEW

File: packages/commonwealth/shared/utils.ts [hunk 1]
```diff
@@ -282,3 +282,34 @@ export function aggregatePermissions(
...
+ * Example 7000000 uosmo -> 7 OSMO
+ * Example 8000000000000000000 aevmos -> 8 EVMOS
+ *
+ */
+export function minimalToNaturalDenom(
+  amount?: string | number,
+  decimals?: number
+): number {
+  if (!amount || !decimals) return 0;
+  const amt = typeof amount === 'string' ? Number(amount) : amount;
```
Comment (jnaviask) [inline packages/commonwealth/shared/utils.ts:297]: Will this number conversion cause overflow / rounding errors at large token values?

File: packages/commonwealth/shared/utils.ts [hunk 2]
```diff
@@ -282,3 +282,34 @@ export function aggregatePermissions(
...
+ * Example 8 EVMOS -> 8000000000000000000 aevmos
+ * Intended for Cosmos use (decimals usually 6 or 18)
+ */
+export function naturalDenomToMinimal(
+  naturalAmount?: string | number,
+  decimals?: number
+): number {
+  if (!naturalAmount) return 0;
+  const amount = Number.parseFloat(naturalAmount.toString()).toFixed(decimals);
+  return +amount * Math.pow(10, decimals);
```
Comment (jnaviask) [inline packages/commonwealth/shared/utils.ts:314]: Also concerned about conversion errors here.

**Review by jnaviask — CHANGES_REQUESTED** [review]
Worried about rounding / numeric errors in conversion code.

File: packages/commonwealth/shared/utils.ts (hunk 1 above)
Comment (mhagel) [reply packages/commonwealth/shared/utils.ts]: You're right, it's not good for big numbers. Will use keplr's utils and add tests

File: packages/commonwealth/shared/utils.ts (hunk 2 above)
Comment (mhagel) [reply packages/commonwealth/shared/utils.ts]: addressed above

File: packages/commonwealth/shared/utils.ts (hunk 1 above)
Comment (mhagel) [reply packages/commonwealth/shared/utils.ts]: Unit tests added. LMK if there are some other cases I should consider. @jnaviask

**Review by jnaviask — APPROVED** [review]

**Pull request merged**